source/air/src/main.rs

```diff
@@ -140,10 +144,16 @@ pub fn main() {
             ValidityResult::UnexpectedSmtOutput(err) => {
                 panic!("Unexpected SMT output: {}", err);
             }
+            ValidityResult::UnexpectedSingularOutput(err) => {
+                panic!("Unexpected Singular output: {}", err);
+            }
         }
         if matches!(**command, CommandX::CheckValid(..)) {
             air_context.finish_query();
         }
+        if matches!(**command, CommandX::SingularCheckValid(..)) {
+            air_context.finish_query();
+        }
```

Comment:
As discussed above, this is redundant with 151-153 due to having a separate CommandX variant.